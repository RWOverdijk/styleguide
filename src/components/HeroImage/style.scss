/** @define HeroImage */

.Van-HeroImage {
  position: relative;
  height: 400px;

  &-content {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background-size: cover;
    background-position: 50% 50%;
    background-repeat: no-repeat;
    z-index: 1;
  }
}
