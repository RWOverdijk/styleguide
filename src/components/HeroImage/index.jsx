import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './style.scss';

class HeroImage extends PureComponent {
  render() {
    const {
      className,
      image,
      ...props
    } = this.props;

    return (
      <div
        className={classnames('Van-HeroImage', className)}
        {...props}
      >
        <div
          className="Van-HeroImage-content"
          style={{
            backgroundImage: `url(${image})`,
          }}
        />
      </div>
    );
  }
}

HeroImage.defaultProps = {
  className: '',
  image: '',
};

HeroImage.propTypes = {
  className: PropTypes.string,
  image: PropTypes.string,
};

export default HeroImage;
