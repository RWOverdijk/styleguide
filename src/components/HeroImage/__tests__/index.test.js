import React from 'react';
import renderer from 'react-test-renderer';
import HeroImage from '../';

describe('<HeroImage>', () => {
  it("renders HeroImage without an image", () => {
    const tree = renderer.create(
      <HeroImage />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders  HeroImage with an image", () => {
    const tree = renderer.create(
      <HeroImage image="my-image-path" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <HeroImage className="my-class" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <HeroImage data-id="0" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
