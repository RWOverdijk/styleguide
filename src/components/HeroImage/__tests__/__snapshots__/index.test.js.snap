// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroImage> renders  HeroImage with an image 1`] = `
<div
  className="Van-HeroImage"
>
  <div
    className="Van-HeroImage-content"
    style={
      Object {
        "backgroundImage": "url(my-image-path)",
      }
    }
  />
</div>
`;

exports[`<HeroImage> renders HeroImage without an image 1`] = `
<div
  className="Van-HeroImage"
>
  <div
    className="Van-HeroImage-content"
    style={
      Object {
        "backgroundImage": "url()",
      }
    }
  />
</div>
`;

exports[`<HeroImage> renders with specific attributes 1`] = `
<div
  className="Van-HeroImage"
  data-id="0"
>
  <div
    className="Van-HeroImage-content"
    style={
      Object {
        "backgroundImage": "url()",
      }
    }
  />
</div>
`;

exports[`<HeroImage> renders with specific className 1`] = `
<div
  className="Van-HeroImage my-class"
>
  <div
    className="Van-HeroImage-content"
    style={
      Object {
        "backgroundImage": "url()",
      }
    }
  />
</div>
`;
